server/src/server/mod.rs

```diff
@@ -0,0 +1,313 @@
+mod client;
+
+use self::client::Client;
+use game::{Game, Input, PlayerId, World};
+use network::{ConnectionId, Event, Message, WebsocketServer};
+use serde_json;
+use std::collections::{BTreeMap, HashMap};
+use std::str;
+use std::thread;
+use std::time::{Duration, Instant};
+
+pub struct Server {
+    network_server: WebsocketServer,
+    clients: HashMap<ConnectionId, Client>,
+    future_inputs: HashMap<u64, InputSet>,
+    frame: u64,
+    game: Game,
+    world: World,
+    next_update_info: UpdateInfo,
+}
+
+#[derive(Default)]
+struct InputSet {
+    inputs: BTreeMap<PlayerId, Input>,
+}
+
+struct UpdateInfo {
+    frame: u64,
+    new_players: Vec<PlayerId>,
+    removed_players: Vec<PlayerId>,
+    inputs: InputSet,
+}
+
+impl UpdateInfo {
+    fn reset(&mut self, frame: u64) {
+        self.frame = frame;
+        self.new_players.clear();
+        self.removed_players.clear();
+        self.inputs.inputs.clear();
+    }
+}
+
+impl Server {
+    pub fn new(network_server: WebsocketServer, game: Game, world: World) -> Server {
+        Server {
+            network_server,
+            clients: HashMap::new(),
+            future_inputs: HashMap::new(),
+            frame: 0,
+            game,
+            world,
+            next_update_info: UpdateInfo {
+                frame: 1,
+                new_players: Vec::new(),
+                removed_players: Vec::new(),
+                inputs: InputSet {
+                    inputs: BTreeMap::new(),
+                },
+            },
+        }
+    }
+
+    pub fn run(&mut self) {
+        let mut last_frame = Instant::now();
```

Comment:
I think it would make sense if `last_frame` was the number of the last frame (or some other identifier of a frame) and not a timestamp. If timestamps are needed, maybe call it `last_frame_time` or something simlar for better readability?

Reply:
We need timestamps here because we want to run at 60 fps, so we do have to actually measure the time. Renamed `{last,next}_frame` to `{last,next}_frame_time`.